diff --git a/cjs/eslint-plugin-react-hooks.development.js b/cjs/eslint-plugin-react-hooks.development.js
index cccbb45430a73e899171dc64e38d240d64427877..e974606e3e0ad9cf8c9bdc7732a74c47abe7773d 100644
--- a/cjs/eslint-plugin-react-hooks.development.js
+++ b/cjs/eslint-plugin-react-hooks.development.js
@@ -97,7 +97,7 @@
       return !1;
     }
     function isComponentName(node) {
-      return "Identifier" === node.type && /^[A-Z]/.test(node.name);
+      return "Identifier" === node.type && /^_?[A-Z]/.test(node.name);
     }
     function isReactFunction(node, functionName) {
       return (
diff --git a/cjs/eslint-plugin-react-hooks.production.js b/cjs/eslint-plugin-react-hooks.production.js
index aa40a2da816086cd15061207a66aaa10e08241cd..865db8a79f79c96246e89513c3d7b4c01c1e75fe 100644
--- a/cjs/eslint-plugin-react-hooks.production.js
+++ b/cjs/eslint-plugin-react-hooks.production.js
@@ -90,7 +90,7 @@ function isHook(node) {
   return !1;
 }
 function isComponentName(node) {
-  return "Identifier" === node.type && /^[A-Z]/.test(node.name);
+  return "Identifier" === node.type && /^_?[A-Z]/.test(node.name);
 }
 function isReactFunction(node, functionName) {
   return (
